source/test/TestApp/PlayFabGroupsTest.cpp

```diff
@@ -0,0 +1,84 @@
+
+#include "TestAppPch.h"
+
+#include "PlayFabGroupsTest.h"
+#include "playfab/PlayFabApiSettings.h"
+#include "playfab/PlayFabSettings.h"
+
+using namespace PlayFab;
+
+namespace PlayFabUnit
+{
+
+    // GROUPS API
+    void PlayFabGroupsTest::GroupsApiTest(TestContext& testContext)
+    {
+        auto settings = std::make_shared<PlayFabApiSettings>();
+
+        auto req = ClientModels::LoginWithCustomIDRequest();
+        req.CustomId = PlayFabSettings::buildIdentifier;
+
+        clientApi->LoginWithCustomID(req, Callback(&PlayFabGroupsTest::GroupsTestLoginCallback), Callback(&PlayFabGroupsTest::GroupsTestSharedFailureCallback), &testContext);
+    }
+
+    void PlayFabGroupsTest::GroupsTestLoginCallback(const ClientModels::LoginResult& result, void* customData)
+    {
+        auto groupApi = std::make_shared<PlayFabGroupsInstanceAPI>(result.authenticationContext);
+        auto req = GroupsModels::CreateGroupRequest();
+        req.CustomTags = std::map<std::string, std::string>();
+
+        // TODO Bug 29786037: this map is required to be filled to not get a 500 error with the CreateGroup api call
+        req.CustomTags.insert(std::pair<std::string, std::string>("One", "Two"));
+
+        req.GroupName = GenerateRandomString();
+        groupApi->CreateGroup(req, Callback(&PlayFabGroupsTest::GroupsTestGroupCallback), Callback(&PlayFabGroupsTest::GroupsTestSharedFailureCallback), customData);
+    }
+
+    void PlayFabGroupsTest::GroupsTestGroupCallback(const GroupsModels::CreateGroupResponse&, void* customData)
+    {
+        TestContext* testContext = static_cast<TestContext*>(customData);
+        testContext->Pass();
+    }
+
+    void PlayFabGroupsTest::GroupsTestSharedFailureCallback(const PlayFabError& error, void* customData)
+    {
+        TestContext* testContext = static_cast<TestContext*>(customData);
+        testContext->Fail("Expected call to succeed but instead got the error: " + error.GenerateErrorReport());
+    }
+
+    void PlayFabGroupsTest::AddTests()
+    {
+        AddTest("CreateGroupApi", &PlayFabGroupsTest::GroupsApiTest);
+    }
+
+    void PlayFabGroupsTest::ClassSetUp()
+    {
+        clientApi = std::make_shared<PlayFabClientInstanceAPI>(PlayFabSettings::staticPlayer);
```

Comment:
Looks like it may have been intended to use the static settings from the test above, but failed to move it here.

Reply:
Likewise, it's better test isolation if this is initialized with a local settings and context, rather than using staticPlayer and staticContext, which can have interference from other tests.